doc/users/interactive.rst

```diff
@@ -1,12 +1,306 @@
-.. _dump-index:
+.. currentmodule:: matplotlib
 
-===================
- Interactive plots
-===================
+.. _mpl-shell:
+
+=====================
+ Interactive Figures
+=====================
+
+.. toctree::
+
+
+When working with data it is often invaluable to be able to
+interact with your plots  In many cases the built in pan/zoom
+and mouse-location tools are sufficient, but if they are not
+you can make use of the Matplotlib event system to build a customized
+data exploration tool.
+
+Matplotlib ships with :ref:`backends <what-is-a-backend>` binding to
+several GUI toolkits (Qt, Tk, Wx, Gtk, OSX, js) and third party
+packages provide bindings to `kivy
+<https://github.com/kivy-garden/garden.matplotlib>`__ and `Jupyter Lab
+<https://github.com/matplotlib/ipympl>`__.  For the figures to be
+"live" the GUI event loop will need to be integrated with your prompt, the
+simplest way is to use IPython (see :ref:`below <ipython-pylab>`).
+
+The `.pyplot` module provides two functions for creating Figures that,
+when an interactive backend is used, that are managed by Matplotlib,
+embedded in GUI windows, and ready for interactive use out of the box
+
+`.pyplot.figure`
+    Creates a new empty `.figure.Figure` or selects an existing figure
+
+`.pyplot.subplots`
+    Creates a new `.figure.Figure` and fills it with a grid of `.axes.Axes`
+
+Matplotlib keeps a reference to all of the open figures created this
+way so they will not be garbage collected.  You can close all off your
+open figures via ``plt.close('all')``.
+
+For discussion of how this works under the hood see:
 
 .. toctree::
-    :maxdepth: 2
+    :maxdepth: 1
 
-    navigation_toolbar.rst
-    shell.rst
+    interactive_guide.rst
     event_handling.rst
+
+
+.. _ipython-pylab:
+
+IPython integration
+===================
+
+We recommend using IPython for an interactive shell.  In addition to
+all of its features (improved tab-completion, magics,
+multiline-editing, etc), it also ensures that the GUI toolkit event
+loop is properly integrated with the command line (see
+:ref:`cp_integration`).  To configure the integration and enable
+:ref:`interactive mode <controlling-interactive>` use the
+``%matplotlib`` magic
+
+.. sourcecode:: ipython
```

Comment:
instead of repeating `.. sourcecode:: ipython` you can probably use https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-highlight